extensions/amp-sidebar/1.0/component.js

```diff
@@ -135,42 +136,93 @@ function SidebarWithRef(
-          } ${side === Side.LEFT ? classes.left : classes.right}`}
-          role="menu"
-          tabindex="-1"
-          hidden={!side}
-          {...rest}
-        >
-          {children}
-        </ContainWrapper>
-        <div
-          ref={backdropRef}
-          onClick={() => close()}
-          part="backdrop"
-          style={backdropStyle}
-          className={`${backdropClassName ?? ''} ${classes.backdrop} ${
-            classes.defaultBackdropStyles
-          }`}
-          hidden={!side}
-        >
-          <div className={classes.backdropOverscrollBlocker}></div>
-        </div>
-      </>
-    )
+    <div className={objstr({[classes.unmounted]: !mounted})}>
+      <ContainWrapper
+        as={Comp}
+        ref={sidebarRef}
+        size={false}
+        layout={true}
+        paint={true}
+        part="sidebar"
+        wrapperClassName={objstr({
+          [classes.sidebar]: true,
+          [classes.defaultSidebarStyles]: true,
+          [classes.left]: side !== Side.RIGHT,
+          [classes.right]: side === Side.RIGHT,
+        })}
+        role="menu"
+        tabindex="-1"
+        hidden={!side}
+        {...rest}
+      >
+        {children}
+      </ContainWrapper>
+      <div
+        ref={backdropRef}
+        onClick={() => close()}
+        part="backdrop"
+        style={backdropStyle}
+        className={objstr({
+          [backdropClassName]: backdropClassName,
+          [classes.backdrop]: true,
+          [classes.defaultBackdropStyles]: true,
+        })}
+        hidden={!side}
+      >
+        <div className={classes.backdropOverscrollBlocker}></div>
+      </div>
+    </div>
   );
 }
 
 const Sidebar = forwardRef(SidebarWithRef);
 Sidebar.displayName = 'Sidebar'; // Make findable for tests.
 export {Sidebar};
+
+/**
+ * @param {!SidebarDef.SidebarToolbarProps} props
+ * @return {PreactDef.Renderable}
+ */
+export function SidebarToolbar({
+  toolbar,
+  'toolbar-target': toolbarTarget,
+  children,
+  ...rest
+}) {
+  const ref = useRef();
+  const [matches, setMatches] = useState(false);
+  const [targetEl, setTargetEl] = useState(null);
+
+  useEffect(() => {
+    const window = ref.current?.ownerDocument?.defaultView;
+    if (!window) {
+      return;
+    }
+
+    const mediaQueryList = window.matchMedia(toolbar);
+    const updateMatches = () => setMatches(mediaQueryList.matches);
+    mediaQueryList.addEventListener('change', updateMatches);
+    setMatches(mediaQueryList.matches);
+    return () => mediaQueryList.removeEventListener('change', updateMatches);
+  }, [toolbar]);
+
+  useEffect(() => {
+    const document = ref.current?.ownerDocument;
+    if (!document) {
+      return;
+    }
+
+    const selector = `#${CSS.escape(toolbarTarget)}`;
+    const newTargetEl = document.querySelector(selector);
+    setTargetEl(newTargetEl);
+  }, [toolbarTarget]);
+
+  return (
+    <>
+      <nav ref={ref} toolbar={toolbar} toolbar-target={toolbarTarget} {...rest}>
```

Comment:
```suggestion
      <nav ref={ref} toolbar={toolbar} toolbarTarget={toolbarTarget} {...rest}>
```

Reply:
Updated!

Reply:
Why is it important for it to be rendered out in kebab case?

Reply:
Html attributes are in kebab case?  So the output would look like this:

```
<nav toolbar="(max-width: 500px)" toolbar-target="toolbar-target">...
```

whereas if we use camel case here it would output like this:
```
<nav toolbar="(max-width: 500px)" toolbartarget="toolbar-target">...
```

Also the first one is aligned with what our `0.1` code would look like :)